### PR TITLE
Add round trip image data path

### PR DESCRIPTION
**Pull request by patkenneally**
* **Tickets addressed:** 000
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This PR adds the ability for Cielim to return a PNG encoded image to Basilisk. The primary changes start in commit 8, where previously the ZMQMultiThreadActor would simply return the vizMessage payload to the simulation data source, this is replaced with a std::variant Query that can contain one of any of the command/query types from Basilisk (Ping, SimUpdate, RequestImage, or BSKError). This allows the Query type to be enqueued and dequeued into the circular queue between the connection thread and the application main thread. This allows commands and query response (of which the returning image is one) to be passed between both threads. The overall development of this mechanism is currently poor and can stand many improvements (as noted below in the further work). The next key commit is commit 9, where the generation and return trip of the image is connected.  

## Verification
This is currently verified by hand where an asteroid flyby Basilisk scenario is run and the returned images are saved and visually inspected. 

## Documentation
A documentation ticket has been created for this work. It will follow shortly.

## Future work
- integrate the main thread's response to dequeued queries with an eventing/callback lambda type of business logic
- redesign the query/command data types

### CODE REVIEW

File: Source/cielim/Private/CielimLoggingMacros.cpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#include "CielimLoggingMacros.h"
+
+DEFINE_LOG_CATEGORY(LogCielim);
```
Comment (thibaudteil) [inline Source/cielim/Private/CielimLoggingMacros.cpp]: missing extra line here?

File: Source/cielim/Private/ZmqConnection/Connector.cpp [hunk 2]
```diff
@@ -159,47 +185,34 @@ void Connector::Listen()
...
-				break;
-			}
-		}
+		// while (this->IsListenerConnected)
+		// {
+			// if (Response.send(this->ReplySocket))
+			// {
+			// 	break;
+			// }
+		// }
```
Comment (thibaudteil) [inline Source/cielim/Private/ZmqConnection/Connector.cpp]: Not sure if you want to leave this for now

**Review by thibaudteil — APPROVED** [review]

File: Source/cielim/Private/CielimLoggingMacros.cpp (hunk 1 above)
Comment (patkenneally) [reply Source/cielim/Private/CielimLoggingMacros.cpp]: New line added.

File: Source/cielim/Private/ZmqConnection/Connector.cpp (hunk 2 above)
Comment (patkenneally) [reply Source/cielim/Private/ZmqConnection/Connector.cpp]: Removed.

**Review by thibaudteil — APPROVED** [review]

**Pull request merged**